CCIT-90: Add module template for platform v3

mvktsk opened the pull request:
Added Platform V3 new module PowerShell template.

mvktsk committed:
CCIT-90: Add module template for platform v3

eMazeika commented on {RepositoryName}/src/{Namespace}.Data/Repositories/MyVC3Module4DbContext.cs:
```diff
@@ -0,0 +1,25 @@
+using EntityFrameworkCore.Triggers;
```
is the filename "MyVC3Module4DbContext.cs" correct?

mvktsk replied:
Fixed

eMazeika commented on {RepositoryName}/src/{Namespace}.Data/{Namespace}.Data.csproj:
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
remove "/src/{Namespace}.Data/packages.config" file

mvktsk replied:
Fixed

eMazeika commented on {RepositoryName}/src/{Namespace}.Web/Module.cs:
```diff
@@ -0,0 +1,51 @@
+            permissionsProvider.RegisterPermissions(ModuleConstants.Security.Permissions.AllPermissions.Select(x =>
+                new Permission()
+                {
+                    GroupName = "{ModuleId}.Web",
```
redundant ".Web"?

mvktsk replied:
Fixed

eMazeika commented on {RepositoryName}/src/{Namespace}.Web/package.json:
```diff
@@ -0,0 +1,18 @@
+{
```
remove "/src/{Namespace}.Web/package-lock.json" file

mvktsk replied:
But in [OrdersModule](https://github.com/VirtoCommerce/vc-module-order/tree/release/3.0.0/src/VirtoCommerce.OrdersModule.Web) repository "package-lock.json" file is exist.
Need more comments on why "package-lock.json" file should be removed.

eMazeika replied:
oh, an update: everyone SHOULD commit package-lock to source control.

eMazeika commented on {RepositoryName}/src/{Namespace}.Core/ModuleConstants.cs:
```diff
@@ -0,0 +1,61 @@
+                public static SettingDescriptor {ModuleId}Enabled = new SettingDescriptor
+                {
+                    Name = "{ModuleId}.{ModuleId}Enabled",
+                    GroupName = "Store|General",
+                    ValueType = SettingValueType.Boolean,
+                    DefaultValue = false
+                };
+
+                public static SettingDescriptor {ModuleId}Password = new SettingDescriptor
+                {
+                    Name = "{ModuleId}.{ModuleId}Password",
+                    GroupName = "Store|Advanced",
```
GroupName should include a variable

mvktsk replied:
Fixed

eMazeika commented on {RepositoryName}/src/{Namespace}.Web/module.manifest:
```diff
@@ -21,14 +21,6 @@
     <assemblyFile>{Namespace}.Web.dll</assemblyFile>
```
VirtoCommerce.Core and VirtoCommerce.Platform.Core are not the same. Comment out 
`<dependency id="VirtoCommerce.Core" version="{CoreModuleVersion}" />`

mvktsk replied:
fixed

eMazeika reviewed (changes requested):
module name doesn't support "."
Data/Caching/ should include {module}CacheRegion.cs file

mvktsk committed:
CCIT-90: Fix PR issues

mvktsk committed:
FIx  Author parameter default value

mvktsk committed:
CCIT-90: Add PlatformNuGetPackageVersion parameter

CoreModuleVersion DataModuleVersion params replaced by
PlatformNuGetPackageVersion parameter

eMazeika commented on README.md:
```diff
@@ -28,12 +28,12 @@ Difference between this script and [Virto Commerce 3.x Module Templates](https:/
 where `-ModuleName` your module name, `-Author` your name and `-CompanyName` your company name. This command will replace constants with their values in all files.
 
-6. `PlatformVersion` and `CoreModuleVersion` parameters have "3.0.0" "3.0.0-rc4-0225" default values respectively. You can skip this parameters if you use default Platform and CoreModule versions
+6. `PlatformVersion` and `PlatformNuGetPackageVersion` parameters have "3.0.0" "3.0.0-rc4-0225" default values respectively. You can skip this parameters if you use default versions
```
this -> these

mvktsk replied:
fixed

mvktsk committed:
CCIT-90: Fix readme.md review issue

mvktsk committed:
CCIT-90: Fix PlatformNuGetPackageVersion issue

eMazeika commented on {RepositoryName}/src/{Namespace}.Web/Scripts/module.js:
```diff
@@ -0,0 +1,42 @@
+        }
+    ])
+    .run(['$rootScope', 'platformWebApp.mainMenuService', 'platformWebApp.widgetService', '$state',
+        function ($rootScope, mainMenuService, widgetService, $state) {
```
remove '$rootScope', it's never used in real modules

mvktsk replied:
fixed

eMazeika replied:
and also remove $rootScope

mvktsk committed:
CCIT-90: Remove $rootScope in module.js

eMazeika commented on {RepositoryName}/src/{Namespace}.Web/Module.cs:
```diff
@@ -0,0 +1,51 @@
+        {
+            // database initialization
+            var configuration = serviceCollection.BuildServiceProvider().GetRequiredService<IConfiguration>();
+            var connectionString = configuration.GetConnectionString("VirtoCommerce.{ModuleId}.Web") ?? configuration.GetConnectionString("VirtoCommerce");
```
remove ".Web

mvktsk replied:
fixed

mvktsk committed:
CCIT-90: Fix PR review issues

mvktsk committed:
CCIT-90: Set module naming in camelCase in JS

mvktsk committed:
CCIT-90: Fix camelCase naming

mvktsk committed:
Fix a null-valued expression error

mvktsk committed:
Add permissions and settings to localizations

Related task CCIT-90

mvktsk committed:
Fix localization file issue
